setup.py

```diff
@@ -36,14 +36,14 @@
 ]
 
 INSTALL_REQUIRES = [
-    'matplotlib >= 3.0.0',
-    'numpy >= 1.15',
+    'matplotlib >= 3.0.0,<4.0.0',
+    'numpy >= 1.15,<2.0.0',
     # exclude pandas==1.0.0 & 1.0.1 for GH142, and 0.24.0 for GH114
-    'pandas >= 0.23.0,!=0.24.0,!=1.0.0,!=1.0.1',
-    'statsmodels >= 0.8.0',
-    'scipy >= 0.19.1',
-    'h5py >= 2.7.1',
-    'pvlib >= 0.7.0, <0.8.0',
+    'pandas >= 0.23.0,!=0.24.0,!=1.0.0,!=1.0.1,<2.0.0',
+    'statsmodels >= 0.8.0,<1.0.0',
+    'scipy >= 0.19.1,<2.0.0',
+    'h5py >= 2.7.1,<3.0.0',
+    'pvlib >= 0.7.0, <0.9.0',
```

Comment:
I'm not sure we need to be restrictive for packages without known issues (that are beyond version 1). Travis will catch a breaking change in a new version of a dependency and let us know. Pandas for example doesn't seem to be restrictive: https://github.com/pandas-dev/pandas/blob/master/setup.py 

The goal of setup.py is to put some guardrails up but leave pip and conda (for example) as much leeway as possible to solve environments.

Reply:
Well I checked astropy and metpy and nobody uses upper limits like this, so maybe you're right.  I'm imagining a scenario where one of our dependencies releases a new major version after we cut this release and when someone goes to install rdtools in a fresh environment, they get strange errors.  And since we already cut the release, there's nothing we can do about it except cut a fresh one with a fix, but that doesn't help people wanting to use the older release.

I guess our docs make a point of recommending that users use `requirements.txt` to make sure everything works.  I'll revert these upper bounds.